### PR TITLE
Expose .msg/.srv/.action to .idl conversion via rosidl translate CLI

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rosidl/issues/565. Depends on https://github.com/ros2/rosidl/pull/575.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `rosidl_adapter`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13829/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8690/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11537/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13914/)

File: rosidl_adapter/rosidl_adapter/cli.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2018 Open Source Robotics Foundation, Inc.
+# Copyright 2018-2021 Open Source Robotics Foundation, Inc.
```
Comment (sloretz) [inline rosidl_adapter/rosidl_adapter/cli.py:1]: No need to update the copyright year. I'm pretty sure we don't usually bump it.

File: rosidl_adapter/rosidl_adapter/cli.py [hunk 2]
```diff
@@ -48,3 +51,57 @@ def convert_files_to_idl(extension, conversion_function, argv=sys.argv[1:]):
...
+        return translated_interface_files
+
+
+class TranslateMsgToIDL(TranslateToIDL):
+
+    input_format = 'msg'
+
+    @property
+    def conversion_function(self):
+        from rosidl_adapter.msg import convert_msg_to_idl
```
Comment (sloretz) [inline rosidl_adapter/rosidl_adapter/cli.py:87]: Why is the import in the function instead of on the module?

**Review by sloretz — COMMENTED** [review]
LGTM with a couple nitpicks and windows CI made happy

File: rosidl_adapter/rosidl_adapter/cli.py (hunk 2 above)
Comment (hidmic) [reply rosidl_adapter/rosidl_adapter/cli.py]: This source file didn't import them directly, so I just went with it. But I don't mind strongly. Moved to the top in https://github.com/ros2/rosidl/pull/576/commits/74e0044bdabae5c6f25ce76c6f591423a26ef4ee.

File: rosidl_adapter/rosidl_adapter/cli.py (hunk 1 above)
Comment (hidmic) [reply rosidl_adapter/rosidl_adapter/cli.py]: I've seen both, but I don't mind strongly. Reverted in https://github.com/ros2/rosidl/pull/576/commits/74e0044bdabae5c6f25ce76c6f591423a26ef4ee.

**Review by ahcorde — APPROVED** [review]

**Comment by hidmic** [comment]
Re-running Windows CI after https://github.com/ros2/rosidl/pull/576/commits/928c1a42882af4afb4866dc719d98825901aa0ea:

* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13936/)

**Comment by hidmic** [comment]
And again, using https://github.com/ament/ament_cmake/pull/327:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13853/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8709/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11561/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13941/)

**Comment by hidmic** [comment]
CI's green and https://github.com/ament/ament_cmake/pull/327 is merged. Going in !

**Pull request merged**